File: python/tools/latn_iast_transliteration_verification/latn_iast_to_gujr_test.py
from __future__ import annotations

from indic_script_converter.latn_iast_to_gujr import (
    GujaratiRomanizationProfile,
    IastToGujaratiDigitPolicy,
    IastToGujaratiOmPolicy,
    IastToGujaratiOptions,
    IastToGujaratiPunctuationPolicy,
    to_gujarati_from_iast,
)
from indic_script_converter.tools.example_latn_iast import transliteration_smoke_samples


def main() -> None:
    print('----------------------------------------------------------------')
    print(' 2. GUJARATI TRANSLITERATOR SAMPLES')
    print('----------------------------------------------------------------')

    default_options = IastToGujaratiOptions(
        punctuation_policy=IastToGujaratiPunctuationPolicy.INDIC_DANDA,
        digit_policy=IastToGujaratiDigitPolicy.CONVERT_TO_SCRIPT,
    )

    for source in transliteration_smoke_samples:
        result = to_gujarati_from_iast(source, default_options)
        print(f'"{source}" -> "{result}"')

    _run_option_tests()
    _run_profile_tests()


def _run_option_tests() -> None:
    print('\n[Gujarati Option: Digits Policy (convertToScript)]')
    digit_options = IastToGujaratiOptions(
        digit_policy=IastToGujaratiDigitPolicy.CONVERT_TO_SCRIPT,
    )
    digit_sample = '12345'
    print(f'  "{digit_sample}" -> "{to_gujarati_from_iast(digit_sample, options=digit_options)}"')

    print('\n[Gujarati Option: Danda Policy (indicDanda)]')
    danda_options = IastToGujaratiOptions(
        punctuation_policy=IastToGujaratiPunctuationPolicy.INDIC_DANDA,
    )
    punctuation_sample = 'End. Double end..'
    print(
        f'  "{punctuation_sample}" -> '
        f'"{to_gujarati_from_iast(punctuation_sample, options=danda_options)}"'
    )

    print('\n[Gujarati Option: OM Policy (useOmSign)]')
    om_options = IastToGujaratiOptions(
        om_policy=IastToGujaratiOmPolicy.USE_OM_SIGN,
    )
    om_sample = 'oṃ namaḥ śivāya'
    print(f'  "{om_sample}" -> "{to_gujarati_from_iast(om_sample, options=om_options)}"')


def _run_profile_tests() -> None:
    print('\n[Gujarati Profiles: strictIast / iso15919Core / extendedIndic]')
    strict = IastToGujaratiOptions(
        profile=GujaratiRomanizationProfile.STRICT_IAST,
    )
    iso = IastToGujaratiOptions(
        profile=GujaratiRomanizationProfile.ISO_15919_CORE,
    )
    extended = IastToGujaratiOptions(
        profile=GujaratiRomanizationProfile.EXTENDED_INDIC,
    )

    for source in ['ṛaka', 'ṛhaṇa', 'ḷa', 'laṛkā', 'xaṇḍa']:
        print(f'  "{source}" (strictIast)    -> "{to_gujarati_from_iast(source, options=strict)}"')
        print(f'  "{source}" (iso15919Core) -> "{to_gujarati_from_iast(source, options=iso)}"')
        print(f'  "{source}" (extendedIndic)-> "{to_gujarati_from_iast(source, options=extended)}"')


if __name__ == '__main__':
    main()

File: javascript/tools/latn_iast_transliteration_verification/latn-iast-to-gujr-test.js
import {
    GujaratiRomanizationProfile,
    IastToGujaratiDigitPolicy,
    IastToGujaratiOptions,
    IastToGujaratiPunctuationPolicy,
    IastToGujaratiOmPolicy,
    toGujaratiFromIast,
    transliterationSmokeSamples,
} from '../../src/index.js';

export function main() {
    console.log('----------------------------------------------------------------');
    console.log(' 2. GUJARATI TRANSLITERATOR SAMPLES');
    console.log('----------------------------------------------------------------');

    const defaultOptions = new IastToGujaratiOptions({
        punctuationPolicy: IastToGujaratiPunctuationPolicy.INDIC_DANDA,
        digitPolicy: IastToGujaratiDigitPolicy.CONVERT_TO_SCRIPT,
    });

    for (const source of transliterationSmokeSamples) {
        console.log(`"${source}" -> "${toGujaratiFromIast(source, defaultOptions)}"`);
    }

    runOptionTests();
    runProfileTests();
}

function runOptionTests() {
    console.log('\n[Gujarati Option: Digits Policy (convertToScript)]');
    const digitOptions = new IastToGujaratiOptions({
        digitPolicy: IastToGujaratiDigitPolicy.CONVERT_TO_SCRIPT,
    });
    const digitSample = '12345';
    console.log(`  "${digitSample}" -> "${toGujaratiFromIast(digitSample, digitOptions)}"`);

    console.log('\n[Gujarati Option: Danda Policy (indicDanda)]');
    const dandaOptions = new IastToGujaratiOptions({
        punctuationPolicy: IastToGujaratiPunctuationPolicy.INDIC_DANDA,
    });
    const punctuationSample = 'End. Double end..';
    console.log(
        `  "${punctuationSample}" -> "${toGujaratiFromIast(punctuationSample, dandaOptions)}"`,
    );

    console.log('\n[Gujarati Option: OM Policy (useOmSign)]');
    const omOptions = new IastToGujaratiOptions({
        omPolicy: IastToGujaratiOmPolicy.USE_OM_SIGN,
    });
    const omSample = 'oṃ namaḥ śivāya';
    console.log(`  "${omSample}" -> "${toGujaratiFromIast(omSample, omOptions)}"`);
}

function runProfileTests() {
    console.log('\n[Gujarati Profiles: strictIast / iso15919Core / extendedIndic]');
    const strict = new IastToGujaratiOptions({
        profile: GujaratiRomanizationProfile.STRICT_IAST,
    });
    const iso = new IastToGujaratiOptions({
        profile: GujaratiRomanizationProfile.ISO_15919_CORE,
    });
    const extended = new IastToGujaratiOptions({
        profile: GujaratiRomanizationProfile.EXTENDED_INDIC,
    });

    for (const source of ['ṛaka', 'ṛhaṇa', 'ḷa', 'laṛkā', 'xaṇḍa']) {
        console.log(`  "${source}" (strictIast)    -> "${toGujaratiFromIast(source, strict)}"`);
        console.log(`  "${source}" (iso15919Core) -> "${toGujaratiFromIast(source, iso)}"`);
        console.log(`  "${source}" (extendedIndic)-> "${toGujaratiFromIast(source, extended)}"`);
    }
}

if (import.meta.url === `file://${process.argv[1]}`) main();
